moped-editor/src/components/ActivityMetrics.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from "react";
+import { useMutation } from "@apollo/client";
+import { LAST_SEEN_MUTATION } from "src/queries/staff";
+import { useUser } from "src/auth/user";
+/**
+ * This wrapper calls the set_last_seen endpoint, which updates the user's
+ * last_seen_date timestamp based on their session ID. this action happens
+ * once each time this component mounts
+ */
+export default function ActivityMetrics({ children }) {
+  const { user } = useUser();
+  const [setLastSeen] = useMutation(LAST_SEEN_MUTATION);
+  useEffect(() => {
+    if (!user) {
+      // this event can fire when the user is not logged in
+      return;
+    }
+    setLastSeen().catch((error) => {
```

Comment:
an error would happen if Hasura bombed, or if we have user de-sync issue where where the user's ID in the session token is not actually in the DB. if we don't catch the error, the whole app breaks 💥

Reply:
Nice, yes, that is a good call. The one thing that I thought about is if we would want to consider a threshold for how often this mutation could fire. I was able to spam the new table in the DB by reloading the app in the browser repeatedly (but I don't think users would be doing that on purpose or anything). 

Something to mull over because I'm not sure if adding an extra API call to check last seen before the mutation is great or anything!

Reply:
agreed—this is a really good question. let's watch this table once we deploy, and if needed i like your suggestion of throttling this based on the `last_seen_date`!

Reply:
I can attest that the whole app breaks :(